Makefile

```diff
@@ -164,3 +165,96 @@ ifneq ($(HAS_WEBAPP),)
 	rm -fr webapp/node_modules
 endif
 	rm -fr build/bin/
+
+# server-debug builds and deploys a debug version of the plugin for your architecture.
+# Then resets the plugin to pick up the changes.
+.PHONY: debug
+debug: server-debug reset
+
+.PHONY: server-debug
+server-debug: server/.depensure
+
+	./build/bin/manifest apply
+	mkdir -p server/dist
+
+ifeq ($(OS),OSX)
+	cd server && env GOOS=darwin GOARCH=amd64 $(GO) build -gcflags "all=-N -l" -o dist/plugin-darwin-amd64;
+else ifeq ($(OS),Linux)
+	cd server && env GOOS=linux GOARCH=amd64 $(GO) build -gcflags "all=-N -l" -o dist/plugin-linux-amd64;
+else ifeq ($(OS),Windows_NT)
+	cd server && env GOOS=windows GOARCH=amd64 $(GO) build -gcflags "all=-N -l" -o dist/plugin-windows-amd64.exe;
+else
+	$(error Please set your OS system env variable to: 'OSX', 'Windows_NT', or 'Linux')
+endif
+
+	cd server && cp -r templates dist/templates
+	rm -rf dist/
+	mkdir -p dist/$(PLUGIN_ID)
+	cp $(MANIFEST_FILE) dist/$(PLUGIN_ID)/
+	mkdir -p dist/$(PLUGIN_ID)/server/dist
+	cp -r server/dist/* dist/$(PLUGIN_ID)/server/dist/
+	mkdir -p ../mattermost-server/plugins
+	cp -r dist/* ../mattermost-server/plugins/
+
+# webapp-debug builds and deploys a debug version of the plugin's webapp
+.PHONY: webapp-debug
+webapp-debug:
+
+ifneq ($(HAS_WEBAPP),)
+	# link the webapp directory
+	mkdir -p ../mattermost-server/plugins/$(PLUGIN_ID)/webapp
+	mkdir -p ./webapp
+	ln -nfs $(PWD)/webapp/dist ../mattermost-server/plugins/$(PLUGIN_ID)/webapp/dist
+	# start an npm watch
+	cd webapp && $(NPM) run run &
+endif
+
+	@echo "\n\n*** After the frontend is compiled, run 'make reset' to reset the plugin. Run reset every time a change is made to force the server to serve the chages in your webapp portion of the plugin.\n\n"
+
+# Reset the plugin
+.PHONY: reset
```

Comment:
There is a bunch of duplication here that should be resolved.

Reply:
Great, thanks. That seems to work on OSX as well, so we'll use that.